packages/devtools_app/lib/src/network/network_request_inspector_views.dart

```diff
@@ -151,6 +160,92 @@ class HttpResponseView extends StatelessWidget {
   }
 }
 
+class ImageResponseView extends StatelessWidget {
+  const ImageResponseView(this.data);
+
+  final DartIOHttpRequestData data;
+
+  @override
+  Widget build(BuildContext context) {
+    final img = image.decodeImage(data.encodedResponse);
+    // Format should be of the form: [image/$format]
+    String format = data.type.split('/')[1];
+    format = format.substring(0, format.length - 1);
+    return Column(
+      crossAxisAlignment: CrossAxisAlignment.stretch,
+      children: [
+        _buildTile(
+          'Image Preview',
+          [
+            Padding(
+              padding: const EdgeInsets.all(8.0),
+              child: Image.memory(
+                data.encodedResponse,
+              ),
+            ),
+          ],
+        ),
+        LayoutBuilder(builder: (context, constraints) {
+          return _buildTile(
+            'Metadata',
+            [
+              _buildRow(
+                context,
+                'Format',
+                format,
+                constraints,
+              ),
+              _buildRow(
+                context,
+                'Size',
+                prettyPrintBytes(
+                  data.encodedResponse.lengthInBytes,
+                  includeUnit: true,
+                ),
+                constraints,
+              ),
+              _buildRow(
+                context,
+                'Dimensions',
+                '${img.width} x ${img.height}',
+                constraints,
+              ),
+            ],
+          );
+        }),
+      ],
+    );
+  }
+
+  Widget _buildRow(
+    BuildContext context,
+    String key,
+    dynamic value,
+    BoxConstraints constraints,
+  ) {
+    return Container(
+      width: constraints.minWidth,
+      padding: _rowPadding,
+      child: Row(
+        crossAxisAlignment: CrossAxisAlignment.start,
+        children: [
```

Comment:
if you use `mainAxisSize: MainAxisSize.min` I think you should be able to get rid of the constraints that are passed around everywhere and remove the wrapping container